decayAmplitude/isobarDecayTopology.cc

```diff
@@ -502,159 +506,125 @@ isobarDecayTopology::getIsospinSymmetrization()
 
-	// Saving all the isospins z-projections.
-	//
+	// Saving the z-projections of all isospins .
 	// Needed to reset everything as it was at the end.
 	vector<int> isospinProjs;
-	for(unsigned int j = 0; j < fsParts.size(); ++j) {
-		isospinProjs.push_back(fsParts[j]->isospinProj());
+	for(unsigned int i = 0; i < fsParts.size(); ++i) {
+		isospinProjs.push_back(fsParts[i]->isospinProj());
 	}
 
 	// A vector to save the found permutations and their Clebsch-Gordans
 	vector<symTermMap> symAmplitudes;
 
 	// Permuting all the particles and checking if the permutation makes sense
-	//
-	const std::vector< std::vector< std::vector<unsigned int> > > permutations = getAllGroupPermutations(groups);
-	for(unsigned int i_permutation = 0; i_permutation < permutations.size(); ++i_permutation){
-		const std::vector< std::vector<unsigned int> >& permutation = permutations[i_permutation];
-		std::vector<unsigned int> map(fsParts.size());
-
-		for(unsigned int i_group = 0; i_group < groups.size(); ++i_group){
-			for(unsigned int i_group_entry = 0; i_group_entry < groups[i_group].size(); ++i_group_entry){
-				map[ groups[i_group][i_group_entry] ] = permutation[i_group][i_group_entry];
+	const vector< vector< vector<unsigned int> > > permutations = getAllGroupPermutations(groups);
+	for(unsigned int i_permutation = 0; i_permutation < permutations.size(); ++i_permutation) {
+		const vector< vector<unsigned int> >& permutation = permutations[i_permutation];
+		vector<unsigned int> permMap(fsParts.size());
+		for(unsigned int i_group = 0; i_group < groups.size(); ++i_group) {
+			for(unsigned int i_groupEntry = 0; i_groupEntry < groups[i_group].size(); ++i_groupEntry) {
+				permMap[ groups[i_group][i_groupEntry] ] = permutation[i_group][i_groupEntry];
 			}
 		}
+		// check that the to-be-added permutation permMap is not equivalent to identity
+		// and not identical to any of the already added permutations
+		// assumes that first permMap is the identity
 		if(_debug) {
-			printDebug << "Processing permutation '";
-			for(unsigned int i = 0; i < map.size(); ++i) cout << map[i];
-			cout << "'." << std::endl;
+			printDebug << "processing permutation " << permMapString(permMap) << endl;
 		}
-		// Some first checks if the permutation makes sense.
-		bool breaking = false;
-		for(std::vector<symTermMap>::const_iterator symAmp = symAmplitudes.begin(); symAmp != symAmplitudes.end(); ++symAmp) {
-			for(unsigned int j = 0; j < map.size(); ++j) {
+		bool discardPermutation = false;
+		for(vector<symTermMap>::const_iterator symAmp = symAmplitudes.begin(); symAmp != symAmplitudes.end(); ++symAmp) {
+
+			if(_debug) {
+				cout << "    ... checking against permutation " << permMapString(symAmp->fsPartPermMap) << endl;
+			}
 
-				if (symAmp->fsPartPermMap[j] == map[j]) {
-					continue; // particle j was not swapped -> go on with check of next particle
+			for(unsigned int i = 0; i < permMap.size(); ++i) {
```

Comment:
Perhaps ‘i’ -> ‘i_particle’ would be better, because the body of the loop is rather long.